### DISCUSSION TITLE
NetBox DNS 1.0 has been released

### DISCUSSION THREAD

**Discussion by peteeckel** [sys4/netbox-plugin-dns#238] [Announcements]
# Release Announcement
I'm pleased to announce that only minutes after the release of NetBox 4.0.0 I could also push NetBox DNS 1.0.0 to PyPI.

## Installation
NetBox DNS 1.0.0 can be installed on systems running NetBox 4.0.0 with the command
```
/opt/netbox/venv/bin/pip install netbox-plugin-dns
```
After the installation, the usual steps (migrate and restart) need to be performed. Otherwise there are no changes.

## New Features
Apart from some improvements in the GUI and some minor changes to keep NetBox DNS functionally and aesthetically in line with NetBox, the functionality is largely identical to NetBox DNS 0.22.8. The major exception is GraphQL, where some improvements where made possible due to the switch from `graphene` to `strawberry`, and the new 'default view' concept.

### New filtering syntax with GraphQL
NetBox moved to a new GraphQL framework with release 4.0.0, so the syntax for GraphQL queries has been changed slightly, especially with filtering. For details, see https://github.com/netbox-community/netbox/blob/feature/docs/integrations/graphql-api.md#filtering.

### New prefix for GraphQL queries
Unlike the REST API, GraphQL does not provide namespaces to provide any means to keep the names of objects unique within its namespace. For that reason I decided to change the query names for NetBox DNS:
| old | new |
| --- | --- |
| view | netbox_dns_view |
| view_list | netbox_dns_view_list |
| nameserver | netbox_dns_nameserver |
| nameserver_list | netbox_dns_nameserver_list |
| zone | netbox_dns_zone |
| zone_list | netbox_dns_zone_list |
| record | netbox_dns_record |
| record_list | netbox_dns_record_list |
| - | netbox_dns_registrar |
| - | netbox_dns_registrar_list |
| - | netbox_dns_contact |
| - | netbox_dns_contact_list |

`contact` and `registrar` have not been available before, among other reasons because `contact` had a name clash with the core NetBox object, which is the main reason for introducing this change.

### Zones without Views are no longer supported
This change makes many parts of the logic easier to handle, and some new features such as an advanced IPAM integration would not be possible with reasonable effort without this new concept. There is an effect to the API, but for general users the impact should be barely noticeable.

See the documentation for more details.

**Comment by bitcollector1** [comment]
Thanks again for the update. 

I forgot to check the helper scripts for updates!  My zones were not updating and I found it was because of the rename of default in my path for the helper script. 

old path
```
ebay@netdevbox:/opt/netbox-dns-exporter$ ls
_default
```

new path
```
ebay@netdevbox:/opt/netbox-dns-exporter$ ls
_default_
```

It only took a couple of minutes to realize what was up as soon as I realized the zones were not updating; it was not apparent from the plugin, everything looked good :)  

It's my fault for not reading the documentation; but there was also a LOT to digest with the change to NetBox 4.0 

Appreciate all that you do for the community!
